src/Arcus.Messaging.Abstractions/MessageCorrelationResult.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using Arcus.Messaging.Abstractions.MessageHandling;
+using GuardNet;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Arcus.Messaging.Abstractions
+{
+    /// <summary>
+    /// Represents the correlation result of a received Azure Service Bus message.
+    /// This result will act as the scope of the request telemetry.
+    /// </summary>
+    public class MessageCorrelationResult : IDisposable
+    {
+        private readonly TelemetryClient _telemetryClient;
```

Comment:
TelemetryClient is an Application Insights specific class, so I think it is a bit weird to see it here in a namespace ethat is called 'Abstractions', and should not be coupled to App Insights ?

Reply:
The Observability library is not coupled to Application Insights, but we already have made that coupling in the W3C HTTP correlation support as Web API (and Messaging) are seen as 'implementations' of Observability.
If we want to include automatic dependency tracking for built-in Microsoft dependencies, without the additional configurations and publicly exposed extensions, and without action from the consumer, we have to include a reference to Application Insights without our 'implementations' of Observability. (like we did in Web API).
Also note that 'abstractions' here means 'messaging abstractions' as in these types are not coupled on a messaging system (message pump or Azure Functions).

Reply:
But using this TelemetryClient class means that we have a dependency on ApplicationInsights in this assembly.   I thought the only place were we had this dependency, was in the ApplicationInsights sink project.

Reply:
No, we can't do that, as there are different Application Insights packages needed to do request/dependency tracking. The Web API needs the `AspNetCore` sub-package and the message pump and all other non-HTTP stuff needs `WorkerService`. The dependency tracking module is available via the dependency collector project, but all necessary boilerplate stuff is around. This module needs the telemetry configuration which means that a registration needs to be configured, this registration can either be done via the AspNetCore or WorkerService package. If we would try to register all the necessary stuff ourselves, we would not only make it very unstable for change, we could also break functionality if people want to extend Application Insights configuration.

Reply:
Yes, I know. It's a shame. Don't see a way to do better than this at the moment.

Reply:
OK. Let's put our quest for purity aside and be pragmatic. I think the value of auto-tracking is high enough to make this concession.